Adding IoT documentation

dejanb opened the pull request:
A first stab at moving readme file into a proper doc

lulf commented on documentation/assemblies/assembly-iot-guide.adoc:
```diff
@@ -0,0 +1,20 @@
+
+include::../modules/proc-installing-iot-services.adoc[leveloffset=+1]
+
+include::../iot/common/proc-creating-iot-project.adoc[leveloffset=+1]
```
These should all be directly under modules (no hierarchy) if i've understood @ncbaratta and @jenmalloy correctly

dejanb replied:
Yeah, I started like that, but figured we'll add a lot of clutter as we continue adding docs in this area.

ncbaratta replied:
I agree @dejanb but the idea is to let anyone use our modules and so all projects have to stick to the same model of putting them all together ... and being cluttered :(

jenmalloy replied:
@dejanb, one way to make clear which modules are specific to IoT might be to use a naming scheme like this: proc-iot-file-name.adoc
@ncbaratta, does this suggestion violate any of our rules?

ncbaratta replied:
That is a-ok!

lulf commented on documentation/iot/master.adoc:
```diff
@@ -0,0 +1,13 @@
+[appendix]
+
+// Revision information
+//include::common/revision-info.adoc[]
```
This file is probably not needed?

ncbaratta replied:
The revision info file?  It's there to put a timestamp on the end of when it was last updated. It's handy.

lulf replied:
I was thinking the iot/master.adoc

ncbaratta replied:
Oh! :)  So, this would be part of the existing guides instead of its own title?

lulf replied:
Creating a dedicated book for iot could also work I guess. I just saw that the assembly-iot-guide was already added to the master-openshift.adoc. I'm not sure how well it would fit in https://enmasse.io/documentation/ as we don't have 'multi-book' support there yet :)

ncbaratta replied:
I'll let @jenmalloy decide when she does her review :)

lulf commented on documentation/master-openshift.adoc:
```diff
@@ -24,6 +24,8 @@ include::assemblies/assembly-service-admin-guide.adoc[leveloffset=+1]
 
 include::assemblies/assembly-tenant-guide.adoc[leveloffset=+1]
 
+include::assemblies/assembly-iot-guide.adoc[leveloffset=+1]
```
Make sure you add this to master-kubernetes.adoc as well.

dejanb replied:
Will do, when it make it work :)

jenmalloy commented:
@dejanb @lulf I can take a look later today (or this evening)

jenmalloy commented on documentation/assemblies/assembly-iot-guide.adoc:
```diff
@@ -0,0 +1,20 @@
+//
+// master-openshift.adoc
+
+:context: {context}-iot
```
@ncbaratta, Should this just be "iot"?

ncbaratta replied:
Depends. Is it going to be a separate book? If not, I don't see a need for a different context at all.

dejanb replied:
I needed new context so that I can include "modules/proc-installing-using-bundle.adoc".

dejanb replied:
Changed title to "Internet of Things (IoT) guide" as we discussed

jenmalloy commented on documentation/iot/common/proc-creating-iot-project.adoc:
```diff
@@ -0,0 +1,29 @@
+
+.Procedure
+
+. Switch back to a non-admin user
```
Change to a non-admin user:

jenmalloy commented on documentation/iot/common/proc-creating-iot-project.adoc:
```diff
@@ -0,0 +1,29 @@
+
+. Switch back to a non-admin user
+
+. Create Managed IoT Project
```
Create a managed IoT project:

dejanb commented on documentation/iot/common/snip-telemetry-consumer.adoc:
```diff
@@ -0,0 +1,25 @@
+. Start telemetry consumer
+In Hono project run
```
I think we'll find a better way to do this, so this is just a temporarily there. I wouldn't bother too much with it for now.

dejanb commented on documentation/modules/proc-installing-iot-services.adoc:
```diff
@@ -0,0 +1,23 @@
+//
+// assembly-installing-openshift.adoc
+
+[id='installing-iot-services-{context}']
```
All other docs use this format of ids. I rather be consistent.

dejanb commented:
@ncbaratta @jenmalloy @lulf I addressed most of the comments and further reorganized the documents. So now, there's no IoT book and I covered both telemetry and event APIs with the same files. All further comments are welcome :)

ncbaratta commented:
An overall comment: I'd like to see IoT spelled out in headings "Internet of Things" we can then abbreviate in the body of the module.

dejanb commented:
@ncbaratta @jenmalloy Changed title to "Internet of Things (IoT) guide" as we discussed

dejanb commented:
Rebased and squashed
